ggshield/core/scan/scannable.py

```diff
@@ -163,7 +181,10 @@ def __init__(self, url: str, content: str, filemode: Filemode = Filemode.FILE):
         self._url = url
         self._path: Optional[Path] = None
         self._content = content
-        self._utf8_encoded_size = None
+
+    def _read_content(self) -> None:
+        if self._content is not None and self._utf8_encoded_size is None:
```

Comment:
`self._content` cannot be None for StringScannable so we can simplify this a bit:

```suggestion
        if self._utf8_encoded_size is None:
```